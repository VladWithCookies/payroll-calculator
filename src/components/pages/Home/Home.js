import { pipe } from 'ramda';
import { useState } from 'react';

import getNetSalary from 'utils/calculations/getNetSalary';
import TaxCalculator from 'components/organisms/TaxCalculator';
import SalaryStats from 'components/organisms/SalaryStats';
import MainLayout from 'components/templates/MainLayout';

export default function Home() {
  const [salary, setSalary] = useState();
  const handleSubmit = pipe(getNetSalary, setSalary);

  return (
    <MainLayout>
      <TaxCalculator onSubmit={handleSubmit}/>
      {salary && <SalaryStats salary={salary} />}
    </MainLayout>
  );
};
