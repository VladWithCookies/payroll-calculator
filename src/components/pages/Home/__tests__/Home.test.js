import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Home from '../Home';

describe('Home', () => {
  it('matches snapshot', () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });

  describe('when user enters valid data', () => {
    it('shows estimated salary', async () => {
      render(<Home />);

      userEvent.type(screen.getByTestId('experience-field'), '20');
      userEvent.selectOptions(screen.getByTestId('location-field'), ['STOCKHOLM']);
      userEvent.selectOptions(screen.getByTestId('year-field'), ['2020']);
      userEvent.click(screen.getByTestId('submit-button'));

      await waitFor(() => {
        expect(screen.getByTestId('stats')).toHaveTextContent('30 960,00 kr');
      });
    });
  });
});
