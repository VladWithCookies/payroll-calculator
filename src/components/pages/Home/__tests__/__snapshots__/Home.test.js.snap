// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<div>
  <div
    class="chakra-container css-0"
  >
    <div
      class="css-0"
    >
      <h1
        class="chakra-heading css-0"
      >
        Payroll Calculator
      </h1>
    </div>
    <form>
      <div
        class="chakra-stack css-0"
      >
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-0"
            for="field-:r0:"
            id="field-:r0:-label"
          >
            Experience
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-0"
              role="presentation"
            >
              *
            </span>
          </label>
          <div
            class="chakra-numberinput css-0"
          >
            <input
              aria-readonly="false"
              aria-required="true"
              aria-valuemax="9007199254740991"
              aria-valuemin="0"
              autocomplete="off"
              autocorrect="off"
              class="chakra-numberinput__field css-0"
              data-testid="experience-field"
              id="field-:r0:"
              inputmode="decimal"
              name="experience"
              pattern="[0-9]*(.[0-9]+)?"
              required=""
              role="spinbutton"
              type="text"
              value=""
            />
            <div
              aria-hidden="true"
              class="css-0"
            >
              <div
                class="css-0"
                role="button"
                tabindex="-1"
              >
                <svg
                  class="chakra-icon css-0"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div
                class="css-0"
                role="button"
                tabindex="-1"
              >
                <svg
                  class="chakra-icon css-0"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-0"
            for="field-:r1:"
            id="field-:r1:-label"
          >
            Profession
          </label>
          <div
            class="chakra-radio-group css-0"
            role="radiogroup"
          >
            <div
              class="chakra-stack css-0"
            >
              <label
                class="chakra-radio css-0"
                data-checked=""
              >
                <input
                  checked=""
                  class="chakra-radio__input"
                  id="radio-:r3:"
                  name="profession"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="radio"
                  value="DEVELOPER"
                />
                <span
                  aria-hidden="true"
                  class="chakra-radio__control css-0"
                  data-checked=""
                />
                <span
                  class="chakra-radio__label css-0"
                  data-checked=""
                >
                  Developer
                </span>
              </label>
              <label
                class="chakra-radio css-0"
              >
                <input
                  class="chakra-radio__input"
                  id="radio-:r4:"
                  name="profession"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="radio"
                  value="TEACHER"
                />
                <span
                  aria-hidden="true"
                  class="chakra-radio__control css-0"
                />
                <span
                  class="chakra-radio__label css-0"
                >
                  Teacher
                </span>
              </label>
              <label
                class="chakra-radio css-0"
              >
                <input
                  class="chakra-radio__input"
                  id="radio-:r5:"
                  name="profession"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="radio"
                  value="CASHIER"
                />
                <span
                  aria-hidden="true"
                  class="chakra-radio__control css-0"
                />
                <span
                  class="chakra-radio__label css-0"
                >
                  Cashier
                </span>
              </label>
            </div>
          </div>
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-0"
            for="field-:r6:"
            id="field-:r6:-label"
          >
            Location/City
          </label>
          <div
            class="chakra-select__wrapper css-0"
          >
            <select
              class="chakra-select css-0"
              data-testid="location-field"
              id="field-:r6:"
              name="location"
            >
              <option
                value="STOCKHOLM"
              >
                Stockholm
              </option>
              <option
                value="GOTHENBURG"
              >
                Gothenburg
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-0"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentColor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-0"
            for="field-:r7:"
            id="field-:r7:-label"
          >
            Income year
          </label>
          <div
            class="chakra-select__wrapper css-0"
          >
            <select
              class="chakra-select css-0"
              data-testid="year-field"
              id="field-:r7:"
              name="year"
            >
              <option
                value="2019"
              >
                2019
              </option>
              <option
                value="2020"
              >
                2020
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-0"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentColor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <button
          class="chakra-button css-0"
          data-testid="submit-button"
          type="submit"
        >
          Calculate Salary
        </button>
      </div>
    </form>
  </div>
</div>
`;
